interop-testing/src/main/java/io/grpc/testing/istio/EchoTestServer.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.testing.istio;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.ForwardingServerCall;
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.InsecureServerCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.Server;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerInterceptors;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.MetadataUtils;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.istio.EchoTestServiceGrpc.EchoTestServiceImplBase;
+import io.grpc.testing.istio.Istio.ForwardEchoRequest;
+import io.grpc.testing.istio.Istio.ForwardEchoResponse;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class EchoTestServer {
+
+  private static Logger logger = Logger.getLogger(EchoTestServer.class.getName());
+
+  /** 3333 is the magic port that the istio testing for k8s health checks. */
+  private static final int ISTIO_K8S_HEALTH_CHECK_PORT = 3333;
+  private static final String REQUEST_ID = "x-request-id";
+  private static final String STATUS_CODE = "StatusCode";
+  private static final String HOST = "Host";
+  private static final String HOSTNAME = "Hostname";
+  private static final String REQUEST_HEADER = "RequestHeader";
+  private static final String IP = "IP";
```

Comment:
Hmmm, I see your point. On the other hand:

- this is a "contract" with the Istio test framework (and also with other echo servers of other languages): any minor spelling differences can break this (as we have seen before). So this serves as a central place to check/make changes similar to string const definitions, say, in C++
- we might end up using the string literals in more than one place using the same constant.

Not changing it for now. Let me know if you feel otherwise

Reply:
I don't feel too strongly, that's pretty much a minor nit.